pkg/pool/pool.go

```diff
@@ -12,4 +12,5 @@ type Pool interface {
 	Add(ctx context.Context, key string, value interface{}) error
 	Get(ctx context.Context, key string) (interface{}, error)
 	Remove(ctx context.Context, key string) error
+	Disconnect(ctx context.Context) error
```

Comment:
Separate PR: I'm not following why does it need to have ctx? Also probably not necessary for it to be an interface either. And probably not worth a separate package.

Reply:
Ok interesting, that makes sense that I could be part of the kad package. I'm not sure if the original intent was for it to be used elsewhere too. In regards to this ctx, I use it in disconnect to call Remove. But, add, get, and remove don't actually use ctx. I can remove them, what do you think?

Reply:
Not kad but rather `node`. Let's do a separate PR for it.

Reply:
How about the ctx stuff?

Reply:
https://github.com/storj/storj/pull/518

Reply:
@egonelbre could you explain ?